components/log/log.c

```diff
@@ -332,6 +334,61 @@ uint32_t ATTR esp_log_early_timestamp(void)
 
 #ifndef BOOTLOADER_BUILD
 
+char* IRAM_ATTR esp_log_system_timestamp(void)
+{
+    static char buffer[15] = {0};
+    static _lock_t bufferLock = 0;
+
+    if (xTaskGetSchedulerState() == taskSCHEDULER_NOT_STARTED) 
+    {
+        uint32_t timestamp = esp_log_early_timestamp();
```

Comment:
Yeh I added this code because `snprintf()` isn't available while it's booting, so when I was using this function for early logging it was crashing because of something to do with newlib not initialised I think.